src/main/java/datawave/microservice/authorization/postgres/PostgresDatawaveUserLookup.java

```diff
@@ -0,0 +1,218 @@
+                return DatawaveUser.ANONYMOUS_USER;
+            }
+            
+            List<String> myAuths = getAuths(connection, personGuid);
+            auths.addAll(myAuths);
+            
+            // Table/list of Datawave roles associated with the user
+            if (postgresDULProperties.isRolesEnabled()) {
+                List<String> myRoles = getRoles(connection, personGuid);
+                roles.addAll(myRoles);
+            }
+            
+        } catch (SQLException e) {
+            logger.warn(e.toString());
+        }
+        
+        long creationTime = System.currentTimeMillis();
+        long expireTime = creationTime + 60000;
+        return new DatawaveUser(dn, userType, "", auths, roles, null, creationTime, expireTime);
+    }
+    
+    protected String getCN(String subjectDN) {
+        logger.info("Pulling CN out from subject DN...");
+        logger.info(subjectDN);
+        logger.info(postgresDULProperties.getCnRegex());
+        Pattern pattern = Pattern.compile(postgresDULProperties.getCnRegex(), Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(subjectDN);
+        
+        boolean matchFound = matcher.find();
+        if (matchFound) {
+            String match = matcher.group("sn");
+            return match;
+        } else {
+            System.out.println("Match not found");
+        }
+        
+        throw new RuntimeException("No 'cn' found within certificate DN, cannot get user info.");
+    }
+    
+    /**
+     * Create the connection with the Postgresql Database
+     * 
+     * @return
+     * @throws SQLException
+     */
+    private Connection getConnection() throws SQLException {
+        logger.info("creating connection to DB...");
+        String url = String.format("jdbc:postgresql://%s/%s", postgresDULProperties.getDbAddr(), postgresDULProperties.getDbName());
+        Properties props = new Properties();
+        props.setProperty("user", postgresDULProperties.getDbUser());
+        props.setProperty("password", postgresDULProperties.getDbPassword());
+        
+        return DriverManager.getConnection(url, props);
+    }
+    
+    /**
+     * Gets the person GUID using the subject CN field of the certificate. Currently comparing to the first name of the user.
+     * 
+     * @param connection
+     * @param dn
+     *            - certificate information
+     * @return The integer representing the personGuid
+     * @throws SQLException
+     */
+    private int getPersonGuid(Connection connection, SubjectIssuerDNPair dn, UserType userType) throws SQLException {
+        logger.info("Getting the person GUID...");
+        // This is intended to add a sub-query string, depending on route being used.
+        String query = buildIdQuery();
+        logger.info(String.format("Database Query:%s", query));
+        
+        // TODO get first name from dn
+        logger.info(String.format("DN Regex:%s", postgresDULProperties.getDnRegex()));
```

Comment:
Missing space between semi-colon and %s. Should have it to match the consistency of the other lines.

Reply:
This one is good to go but i hadn't noticed theres another missing space 3 lines up!